mesonbuild/mtest.py

```diff
@@ -61,6 +61,23 @@
 # Exit if 3 Ctrl-C's are received within one second
 MAX_CTRLC = 3
 
+# Define unencodable xml characters' regex for replacing
+# them with their printable representation
+unencodable_unichrs = [(0x00, 0x08), (0x0B, 0x0C), (0x0E, 0x1F), (0x7F, 0x84),
+                            (0x86, 0x9F), (0xFDD0, 0xFDEF), (0xFFFE, 0xFFFF)]
+# Not narrow build
+if sys.maxunicode >= 0x10000:
+    unencodable_unichrs.extend([(0x1FFFE, 0x1FFFF), (0x2FFFE, 0x2FFFF),
+                                (0x3FFFE, 0x3FFFF), (0x4FFFE, 0x4FFFF),
+                                (0x5FFFE, 0x5FFFF), (0x6FFFE, 0x6FFFF),
+                                (0x7FFFE, 0x7FFFF), (0x8FFFE, 0x8FFFF),
+                                (0x9FFFE, 0x9FFFF), (0xAFFFE, 0xAFFFF),
+                                (0xBFFFE, 0xBFFFF), (0xCFFFE, 0xCFFFF),
+                                (0xDFFFE, 0xDFFFF), (0xEFFFE, 0xEFFFF),
+                                (0xFFFFE, 0xFFFFF), (0x10FFFE, 0x10FFFF)])
+unencodable_ranges = [fr'{chr(low)}-{chr(high)}' for (low, high) in unencodable_unichrs]
+unencodable_xml_chars_re = re.compile('([' + ''.join(unencodable_ranges) + '])')
```

Comment:
I think I would prefer if these were ALL caps variable names, but other than that I don't see any issues.